### PR TITLE
Add a loopback path instead of localhost

### PR DESCRIPTION
**Pull request by Hoooao**
Check for loopback before sending. Used a dispatch map to deal with generic broadcastToReplica. Fixed redundant broadcast of fallbackTrigger.

### CODE REVIEW

File: processes/replica/replica.cc [hunk 1]
```diff
@@ -296,8 +293,6 @@ void Replica::handleMessage(MessageHeader *hdr, byte *body, Address *sender)
         if (fallbackTriggerMsg.has_proof()) {
             // TODO verify proof
             LOG(INFO) << "Fallback trigger has a proof, starting fallback!";
-
```
Comment (Hoooao) [inline processes/replica/replica.cc:299]: This was confusing, i don't think it needs to broadcast this again

File: processes/replica/replica.cc [hunk 2]
```diff
@@ -296,8 +293,6 @@ void Replica::handleMessage(MessageHeader *hdr, byte *body, Address *sender)
...
-            broadcastToReplicas(fallbackTriggerMsg, FALLBACK_TRIGGER);
```
Comment (Hoooao) [inline processes/replica/replica.cc:300]: This was confusing, i don't think it needs to broadcast this again

Comment (dqian3) [reply processes/replica/replica.cc:300]: We do want this, otherwise a client can trigger fallback selectively.

**Comment by dqian3** [comment]
Another thing, with this change, we can get rid of the ugly (and rather confusing) way which we setup a loopback NNG connection on the replicas. Can you make that part of the PR?

Specifically, it would in `getReplicaAddrs` of `config_util.h` and the constructor of Replica.

**Review by dqian3 — APPROVED** [review]

**Pull request merged**